src/pip/_vendor/vendor.txt

```diff
@@ -3,7 +3,7 @@ colorama==0.4.6
 distlib==0.3.6
 distro==1.8.0
 msgpack==1.0.5
-packaging==21.3
+packaging==23.1
```

Comment:
@sbidoul , can you bump packaging to 23.2 please ? (just released by @pradyunsg) 

```suggestion
packaging==23.2
```

Reply:
Upgraded packaging to 23.2, but now some other tests fail with errors like this:

```
pip._vendor.packaging.requirements.InvalidRequirement: Expected package name at the start of dependency specifier
    
pkga@file:///tmp/pytest-of-sbi-local/pytest-6/test_new_resolver_direct_url_e0/workspace/scratch/pkga-1-py2.py3-none-any.whl?a=1&b=2
    ^
```

Is that a regression in `packaging`, because that requirements looks fine to me?

Reply:
> Upgraded packaging to 23.2, but now some other tests fail with errors like this

Are you sure it's related to the 23.2 update, I'm seeing the same kind of issues with a 23.1 run:
https://github.com/pypa/pip/actions/runs/6370876705/job/17292444724#step:8:2095

I just did a quick check and starting with 22.0, packaging is failing to parse Requirements starting with `\n`.

Reply:
Yep it's something in our test wheel generator... looking into it atm.

Reply:
... wow, that's not something I expected to ever get passed into `packaging.requirements.Requirement`.

Reply:
It turns out `email.message.EmailMessage.as_bytes()` tries to wrap headers that are longer than 78 characters and, for some reason, when retrieving the headers, the leading `\n` is preserved, tripping the new requirement parser.

In other words, it seems that

```Requires-Dist:
  pkga@file:///....
```

was accepted before and is not anymore.

I have pushed a commit to "fix" our test suite, but I wonder if there are such `METADATA` files in the wild.

Reply:
... or maybe we should improve our METADATA parser.

Reply:
> ... or maybe we should improve our METADATA parser.

I did that, it is safer.